rts/src/main/java/eta/runtime/RuntimeOptions.java

```diff
@@ -0,0 +1,74 @@
+package eta.runtime;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+public class RuntimeOptions {
+    private Properties p;
+
+    public RuntimeOptions(String path) {
+        load_properties(path);
+    }
+
+    /*
+     * Loads properties from resource which
+     * may be override by system properties
+     */
+    private void load_properties(String path) {
+        Properties lp = new Properties();
+        // Load properties from resource
+        try (InputStream in = getClass().getClassLoader().getResourceAsStream(path)) {
+            if (in != null) lp.load(in);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        Properties sp = System.getProperties();
+        // Set loaded properties to be override by System properties
+        for (String key : lp.stringPropertyNames()) {
+            if (sp.getProperty(key) == null) {
+                sp.setProperty(key, sp.getProperty(key));
+            }
+        }
+        p = sp;
+    }
+
+    public Properties getProperties() {
+        return p;
+    }
+
+    /*
+     * Get and parse value from properties.
+     * Return default value if failed.
+     */
+    private Object getValue(String key, String type, Object d) {
```

Comment:
Rather than using a generic `getValue` method, why not put the specific code inside the respective `getInt`, `getBoolean`, `getDouble` functions? Yes, there will be repetition but I think it'll be a bit clearer and avoids unnecessary boxing for primitive types.